library/spdm_requester_lib/libspdm_req_key_exchange.c

```diff
@@ -536,6 +536,19 @@ static libspdm_return_t libspdm_try_send_receive_key_exchange(
                 status = LIBSPDM_STATUS_INVALID_MSG_FIELD;
                 goto receive_done;
             }
+
+            /* relax the key_usage_bit check for slot 0, because the responder might no know the key usage yet */
```

Comment:
If the Responder has chosen `SPDM_KEY_EXCHANGE_RESPONSE_MUT_AUTH_REQUESTED` then it should already know the Requester's multi-key restrictions, else the Responder would issue `GET_DIGESTS`. So the `(*req_slot_id_param != 0x0)` is not needed.

Reply:
This field is mandatory.

Without the check, how do you know if this field should be used or not?

Reply:
`SPDM_KEY_EXCHANGE_RESPONSE_MUT_AUTH_REQUESTED` means the Responder wants to skip the encapsulated flow and go straight to `FINISH`. If Responder specifies slot 0 then it means slot 0 and the Requester should check if its slot 0 key can do key exchange.

Reply:
Ah, I see. Agree.